test/parsers/JsonUnitLinkEditBodyParserSpec.scala

```diff
@@ -0,0 +1,77 @@
+package parsers
+
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ EitherValues, FreeSpec, Matchers }
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.http.Status.{ BAD_REQUEST, UNSUPPORTED_MEDIA_TYPE }
+import play.api.test.FakeRequest
+import play.mvc.Http.HeaderNames.CONTENT_TYPE
+import play.mvc.Http.MimeTypes.{ JSON, XML }
+import uk.gov.ons.sbr.models._
+
+/*
+ * See https://github.com/playframework/playframework/blob/master/framework/src/play/src/test/scala/play/mvc/RawBodyParserSpec.scala
+ * for an example of how a BodyParser can be tested.
+ */
+class JsonUnitLinkEditBodyParserSpec extends FreeSpec with Matchers with GuiceOneAppPerSuite with ScalaFutures with EitherValues {
+
+  private trait Fixture {
+    implicit val materializer = app.materializer
+
+    val FromParentLEU = "123456789"
+    val ToParentLEU = "234567890"
+    val Username = "abcd"
+
+    val VATEditParentLinkPostBody =
+      s"""{
+          |  "parent": {
+          |    "from": {
+          |      "id": "$FromParentLEU",
+          |      "type": "LEU"
+          |    },
+          |    "to": {
+          |      "id": "$ToParentLEU",
+          |      "type": "LEU"
+          |    }
+          |  },
+          |  "audit": { "username": "$Username" }
+          |}""".stripMargin
+  }
+
+  "A body representing a JSON patch specification" - {
+    "can be parsed when valid" in new Fixture {
+      val request = FakeRequest().withHeaders(CONTENT_TYPE -> JSON)
+      val body = Source.single(ByteString(VATEditParentLinkPostBody))
+
+      whenReady(JsonUnitLinkEditBodyParser.apply(request).run(body)) { result =>
+        val from = IdAndType(UnitId(FromParentLEU), UnitType.LegalUnit)
```

Comment:
Should we move the definition of the expected result into the fixture?  It is dependent upon the json body, which is defined there.